url/IdnaTestV2.window.js

```diff
@@ -0,0 +1,27 @@
+promise_test(() => fetch("resources/IdnaTestV2.json").then(res => res.json()).then(runTests), "Loading data…");
+
+function runTests(idnaTests) {
+  for (const idnaTest of idnaTests) {
+    if (typeof idnaTest === "string") {
+      continue // skip comments
+    }
+    if (idnaTest.input === "") {
+      continue // cannot test empty string input through new URL()
+    }
+    // Percent-encode the input such that ? and equivalent code points do not end up counting as
+    // part of the URL, but are parsed through the host parser instead.
+    const encodedInput = encodeURIComponent(idnaTest.input);
```

Comment:
Should you maybe just escape the characters that cause state changes instead, like ? and #? It seems interesting to test the host parser on Unicode inputs instead of only ASCII ones.